src/components/application_manager/test/commands/mobile/on_system_request_notification_test.cc

```diff
@@ -199,6 +200,89 @@ TEST_F(OnSystemRequestNotificationTest, Run_RequestNotAllowed_NoNotification) {
   command->Run();
 }
 
+TEST_F(
+    OnSystemRequestNotificationTest,
+    Run_RequestTypeAllowedAndRequestSubTypeDisallowed_MessageNotSentToMobile) {
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+  MockPolicyHandlerInterface mock_policy_handler;
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(ReturnRef(mock_policy_handler));
+  const std::string policy_app_id = "fake-app-id";
+  ON_CALL(*mock_app, policy_app_id()).WillByDefault(Return(policy_app_id));
+
+  const mobile_apis::RequestType::eType request_type =
+      mobile_apis::RequestType::HTTP;
+  EXPECT_CALL(mock_policy_handler,
+              IsRequestTypeAllowed(policy_app_id, request_type))
+      .WillOnce(Return(true));
+  const std::string request_subtype = "fakeSubType";
+  EXPECT_CALL(mock_policy_handler,
+              IsRequestSubTypeAllowed(policy_app_id, request_subtype))
+      .WillOnce(Return(false));
+
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(_, _)).Times(0);
+
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+  (*msg)[strings::msg_params][strings::request_type] = request_type;
+  (*msg)[am::strings::msg_params][am::strings::request_subtype] =
+      request_subtype;
+
+  SharedPtr<OnSystemRequestNotification> command =
+      CreateCommand<OnSystemRequestNotification>(msg);
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(OnSystemRequestNotificationTest,
+       Run_RequestTypeAllowedAndRequestSubTypeAllowed_SendMessageToMobile) {
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app));
+  MockPolicyHandlerInterface mock_policy_handler;
+  ON_CALL(app_mngr_, GetPolicyHandler())
+      .WillByDefault(ReturnRef(mock_policy_handler));
+  const std::string policy_app_id = "fake-app-id";
+  ON_CALL(*mock_app, policy_app_id()).WillByDefault(Return(policy_app_id));
+
+  const mobile_apis::RequestType::eType request_type =
+      mobile_apis::RequestType::OEM_SPECIFIC;
+  EXPECT_CALL(mock_policy_handler,
+              IsRequestTypeAllowed(policy_app_id, request_type))
+      .WillOnce(Return(true));
+  const std::string request_subtype = "fakeSubType";
+  EXPECT_CALL(mock_policy_handler,
+              IsRequestSubTypeAllowed(policy_app_id, request_subtype))
+      .WillOnce(Return(true));
+
+  smart_objects::SmartObjectSPtr result;
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(_, _))
+      .WillOnce((SaveArg<0>(&result)));
+
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+  (*msg)[strings::msg_params][strings::request_type] = request_type;
+  (*msg)[am::strings::msg_params][am::strings::request_subtype] =
+      request_subtype;
+
+  SharedPtr<OnSystemRequestNotification> command =
```

Comment:
@BSolonenko Same as above

Reply:
@ValeriiMalkov I don't see in this any sense

Reply:
@BSolonenko Please, use C++11 https://adc.luxoft.com/confluence/pages/viewpage.action?pageId=307580753